engine/app/components/citizens_advice_components/input.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class Input < Base
+    attr_reader :name, :label, :error_message, :hint, :width, :value
+
+    def initialize(name:, label:, type:, options: nil)
+      super
+      @name = name
+      @label = label
+      @type = type
+
+      set_options(options)
```

Comment:
Curious why this is a separate hash? Feels like these could all still be named arguments? Possibly missing your intent though.

Reply:
Lots of (>4) params is a bit of a code smell and also enforced by rubocop.  Option object params is one of the refactoring approaches (https://refactoring.guru/introduce-parameter-object), felt like it made sense here since they are also all optional.

Reply:
Ooh interesting. I've always interpreted Introduce Parameter Object more for when you've got like parameters being passed together, e.g. passing `start` and `end` when you really want a `range`. I'd argue in this case we've only moved the arguments around, the component still has the same number of arguments we've just put them in a hash. But I don't feel strongly here, and the distinction between required and optional parameters seems reasonable here. Totally your call 👍